src/shared/components/SqlMonacoEditor.tsx

```diff
@@ -38,11 +50,31 @@ const SqlEditorMonaco: FC<Props> = ({
   autofocus,
   wrapLines,
 }) => {
+  const dispatch = useDispatch()
+  const connection = useRef<ConnectionManager>(new ConnectionManager())
+  const sessionStore = useContext(PersistanceContext)
   const [editor, setEditor] = useState(null)
   const wrapperClassName = classnames('qx-editor--monaco', {
     'qx-editor--monaco__autogrow': autogrow,
   })
 
+  useEffect(() => {
+    if (connection.current) {
+      connection.current.onSchemaSessionChange(
+        sessionStore.selection,
+        sessionStore.setSelection,
+        dispatch,
+        sessionStore.range
+      )
+    }
+  }, [
+    connection.current,
+    sessionStore?.selection,
+    sessionStore?.selection.composition || null,
+    sessionStore?.setSelection,
+    sessionStore.range,
+  ])
```

Comment:
Nice catch.
The `sessionStore.range` should absolutely have the existence check too in the memoization.

The choice of using the existence check in the function call, is not required. Since this function is called when the sessionStore changes from undefined into existing. But it doesn't hurt to add it. Doing so.